hash_table.py

```diff
@@ -0,0 +1,31 @@
+class HashTable(object):
+
+    def __init__(self, size):
+        self.size = size
+        self.table = [[] for i in xrange(size)]
+
+    def get(self, key):
+        index = self.hash(key)
+        for item in self.table[index]:
+            if key == item[0]:
+                return item[1]
+        raise ValueError("Key not in table")
+
+    def set(self, key, val):
+        try:
+            index = self.hash(key)
+        except TypeError:
+            raise
```

Comment:
Since you just re-raise here without doing anything, it might be better not to bother with the try/except.